From 3d70303d0164e1aba6231380658d0787ac07f485 Mon Sep 17 00:00:00 2001
From: Amit Shah <amit.shah@redhat.com>
Date: Mon, 18 Aug 2014 10:00:27 -0400
Subject: [hwrng] hw_random: fix sparse warning (NULL vs 0 for pointer)

Message-id: <e1890743c35bdba3947a2192aa666c7b611f6d96.1408347166.git.amit.shah@redhat.com>
Patchwork-id: 87841
O-Subject: [RHEL7.1 kernel PATCH 4/6] hw_random: fix sparse warning (NULL vs 0 for pointer)
Bugzilla: 1129195
RH-Acked-by: Amos Kong <akong@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Herbert Xu <herbert.xu@redhat.com>

From: Torsten Duwe <duwe@lst.de>

Signed-off-by: Torsten Duwe <duwe@suse.de>
Signed-off-by: Theodore Ts'o <tytso@mit.edu>
(cherry picked from commit 9dda727d37ff6ce1628dd2b8557ad011c49db377)
Signed-off-by: Amit Shah <amit.shah@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/char/hw_random/core.c b/drivers/char/hw_random/core.c
index e8b8d80..f9ae152 100644
--- a/drivers/char/hw_random/core.c
+++ b/drivers/char/hw_random/core.c
@@ -359,7 +359,7 @@ static int hwrng_fillfn(void *unused)
   add_hwgenerator_randomness((void *)rng_fillbuf, rc,
         (rc*current_quality)>>10);
  }
- hwrng_fill = 0;
+ hwrng_fill = NULL;
  return 0;
 }
 
-- 
1.7.1